pkg/kubelet/kubelet_node_status.go

```diff
@@ -372,6 +372,19 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 		if err != nil {
 			return fmt.Errorf("failed to get node address from cloud provider: %v", err)
 		}
+
+		if kl.nodeIP != nil {
+			for _, nodeAddress := range nodeAddresses {
+				if nodeAddress.Address == kl.nodeIP.String() {
```

Comment:
Would you just add this node-ip to the list of addresses even if its not part of the list?
The reason i am searching through the list of returned IPs is to make sure that the node-ip can be routable to the node.